FirebaseAppCheck/Sources/AppAttestProvider/FIRAppAttestProvider.m

```diff
@@ -172,6 +174,18 @@ - (void)getTokenWithCompletion:(void (^)(FIRAppCheckToken *_Nullable, NSError *_
 }
 
 - (FBLPromise<FIRAppCheckToken *> *)getToken {
+  return [FBLPromise onQueue:self.queue
+                          do:^id _Nullable {
+                            if (self.ongoingGetTokenOperation == nil) {
+                              // Kick off a new handshake sequence only when there is no an ongoing
+                              // handshake to avoid race conditions.
+                              self.ongoingGetTokenOperation = [self createGetTokenSequencePromise];
+                            }
+                            return self.ongoingGetTokenOperation;
+                          }];
```

Comment:
Question,  should `ongoingGetTokenOperation` get reset back to `nil` once the promise assigned to it is resolved?

Reply:
Good catch! yes, it must be reset and tests should be added for it.

Reply:
Updated here and in FIRAppCheck, PTAL.